http_service/bugbug_http/readthrough_cache.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import datetime
+import logging
+import threading
+import time
+from datetime import timedelta
+from typing import Callable, Dict, Generic, TypeVar
+
+LOGGER = logging.getLogger()
+
+# A simple TTL cache to use with models. Because we expect the number of models
+# in the service to not be very large, simplicity of implementation is
+# preferred to algorithmic efficiency of operations.
+#
+# Called an 'Idle' TTL cache because TTL of items is reset after every get
+Key = TypeVar("Key")
+Value = TypeVar("Value")
+
+
+class ReadthroughTTLCache(Generic[Key, Value]):
+    def __init__(self, ttl: timedelta, load_item_function: Callable[[Key], Value]):
+        self.ttl = ttl
+        self.load_item_function = load_item_function
+        self.items_last_accessed: Dict[Key, datetime.datetime] = {}
+        self.items_storage: Dict[Key, Value] = {}
+        self.last_purged_time = datetime.datetime.now()
+
+    def __contains__(self, key):
+        return key in self.items_storage
+
+    def get(self, key, **kwargs):
+        store_item = kwargs.get("force_store", False)
+        item = None
```

Comment:
Nit: No need to define it as None, as long as a value is given to item in both branches next